jax/experimental/checkify.py

```diff
@@ -212,26 +217,71 @@ def checkify_fun_to_jaxpr(f, error, in_avals):
 
 ## assert primitive
 
-def assert_(pred: Bool, msg: str) -> None:
+def assert_that(pred: Bool, msg: str) -> None:
+  """Assert a predicate, add an error with msg if predicate evaluates to False.
+
+  The predicate needs to be a jnp boolean.
+
+  Args:
+    pred: jnp bool
+    msg: msg to print if assertion fails.
+  Returns:
+    Nothing: this is an effectful operation, and needs to be functionalized
```

Comment:
Technically it returns `None`, since Python functions always have a return value.

Maybe instead of having the description about how to functionalize, we can just provide a link to a docs page which has all the info.

Reply:
yep, I thought it might be a nice place for discussing the fact that it's an effect that needs to be discharged (because my linter was complaining about the `Returns` section missing), but it's just an experiment. Are you thinking about a "checkify" doc to link to?